.github/workflows/push-master.yml

```diff
@@ -66,3 +67,141 @@ jobs:
           destination-repository-name: 'IBAMR-docs'
           user-email: 'travis@travis-ci.org'
           target-branch: master
+
+  # build IBAMR with CMake and update the cache
+  build_ibamr_cmake_016:
```

Comment:
While this should all work (this is ported from Cardinal, where we have been using this approach for a year) there isn't a good way to test this except to merge this PR, wait, open another PR, and verify that we pick up the correct cache.